core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -561,36 +560,26 @@ long applyWAL(
     @Override
     protected boolean doRun(int workerId, long cursor, RunStatus runStatus) {
         final TableToken tableToken;
-        final long seqTxn;
 
         try {
             WalTxnNotificationTask task = queue.get(cursor);
             tableToken = task.getTableToken();
-            seqTxn = task.getTxn();
         } finally {
             // Don't hold the queue until the all the transactions applied to the table
             subSeq.done(cursor);
         }
 
-        final int tableId = tableToken.getTableId();
-        if (lastAppliedSeqTxns.get(tableId) < seqTxn) {
-            // Check, maybe we already processed this table to higher txn.
-            final long txn = applyWAL(tableToken, engine, operationCompiler, runStatus);
-            if (txn > -1L) {
-                lastAppliedSeqTxns.put(tableId, txn);
-            } else if (txn == WAL_APPLY_FAILED) {
-                // Set processed transaction marker as Long.MAX_VALUE - 1
-                // so that when the table is unsuspended it's notified with transaction Long.MAX_VALUE
-                // and is picked up for processing by this apply job.
-                lastAppliedSeqTxns.put(tableId, Long.MAX_VALUE - 1);
-                try {
-                    engine.getTableSequencerAPI().suspendTable(tableToken);
-                } catch (CairoException e) {
-                    LOG.critical().$("could not suspend table [table=").$(tableToken.getTableName()).$(", error=").$(e.getFlyweightMessage()).I$();
-                }
+        // Check, maybe we already processed this table to higher txn.
+        final long txn = applyWAL(tableToken, engine, operationCompiler, runStatus);
+        if (txn == WAL_APPLY_FAILED) {
```

Comment:
nit: since we only check `applyWAL() == WAL_APPLY_FAILED`, we may simplify `applyWAL` to return a `boolean` flag.

Reply:
I'd keep returning max applied transaction for now, perhaps can refactor it later when the dust settles.

Reply:
OK, let's keep it for now.